src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -299,8 +299,9 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
 
   double performance = 0;
   size_t responseSeq = 0;
+  const size_t effectiveRho = (rho > responses.size()) ? responses.size() : rho;
```

Comment:
As per your suggestion
https://github.com/mlpack/mlpack/pull/2086#discussion_r352251518
I tried to do it but it is not working so setting up value like this.

Reply:
`const size_t effectiveRho = std::min(rho, responses.size())`

This doesn't work because response is cube and .size() return uword, you should probably typecast it
`const size_t effectiveRho = std::min(rho, size_t(responses.size()))`

Reply:
Thanks @jeffin143 that helped